crypto/rsa/rsa_oaep.c

```diff
@@ -244,15 +243,16 @@ int RSA_padding_check_PKCS1_OAEP_mgf1(unsigned char *to, int tlen,
      * should be noted that failure is indistinguishable from normal
      * operation if |tlen| is fixed by protocol.
      */
-    tlen = constant_time_select_int(constant_time_lt(dblen, tlen), dblen, tlen);
+    tlen = constant_time_select_int(constant_time_lt(dblen - mdlen - 1, tlen),
+                                    dblen - mdlen - 1, tlen);
     msg_index = constant_time_select_int(good, msg_index, dblen - tlen);
     mlen = dblen - msg_index;
-    for (from = db + msg_index, mask = good, i = 0; i < tlen; i++) {
-        unsigned int equals = constant_time_eq(i, mlen);
+    for (mask = good, i = 0; i < tlen; i++) {
+        unsigned int equals = constant_time_eq(msg_index, dblen);
 
-        from -= dblen & equals; /* if (i == dblen) rewind   */
-        mask &= mask ^ equals;  /* if (i == dblen) mask = 0 */
-        to[i] = constant_time_select_8(mask, from[i], to[i]);
+        msg_index -= (dblen - mdlen - 1) & equals;  /* rewind at EOF */
```

Comment:
I'm wondering if we should use `tlen & equals` instead of `(dblen - mdlen - 1) & equals`. It would always access the same last tlen bytes in that case.

Reply:
I was unsure about that too.  But since tlen is at this point always dblen - mdlen -1
since the constant time expression above does always evaluate the left hand expression,

I found it easier to see this way that the msg_idx jumps from dblen to mdlen + 1

Reply:
It is a bit unfortunate, that tlen exists as a parameter in the padding check functions
and complicates the control flow there, but is not accessible in the exported interface.

Reply:
So I'll change that to `tlen & equals`